lib/cursor/QueryCursor.js

```diff
@@ -104,7 +104,7 @@ QueryCursor.prototype._read = function() {
           return _this.emit('error', error);
         }
         // for node >= 12 the autoDestroy will emit the 'close' event
-        if (utils.nodeMajorVersion < 12) {
+        if (utils.nodeMajorVersion() < 12) {
```

Comment:
If we make `nodeMajorVersion()` return `null` if `process.versions` is undefined, add a check: `const nodeMajorVersion = utils.nodeMajorVersion(); if (nodeMajorVersion != null && nodeMajorVersion < 12) { ... }`

Reply:
Are we sure that in the browser scenario any of the variants (entering or not entering the branch) is correct? Shouldn't it be an error if the execution path reached this point in the browser? I understand we don't to crash too much, but this check may create a false appearance that it can still work.
However, to fully assess that I would need to understand if there is any scenario where QueryCursor can work in the browser.

Reply:
@vkarpov15 What do you think about that? :arrow_up:

Reply:
"Shouldn't it be an error if the execution path reached this point in the browser?" <-- sure, but we can say the same thing about a lot of the codebase. I'd prefer to avoid having individual snippets of code being aware about browser vs not browser, and instead rely on the browser build to determine what should or shouldn't be in the browser build. Ideally this code wouldn't be in the browser build at all, but that's a little harder to fix.

TLDR; let's do the simple, low-risk fix of checking if `nodeMajorVersion` is `null` here.